src/goto-instrument/replace_function_bodies.h

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************\
+
+Module: Replace bodies of goto functions
+
+Author: DiffBlue Limited.
+
+\*******************************************************************/
+
+#ifndef CPROVER_GOTO_INSTRUMENT_REPLACE_FUNCTION_BODIES_H
+#define CPROVER_GOTO_INSTRUMENT_REPLACE_FUNCTION_BODIES_H
+
+#include <regex>
+#include <vector>
+#include <iostream>
+
+#include <goto-programs/goto_functions.h>
+#include <util/std_code.h>
+#include <util/std_types.h>
+#include <util/message.h>
+
+/// Replace function bodies with some default behavior
+/// Can generate the following types of function bodies:
+///
+/// assert-false: { __CPROVER_assert(false); }
+///
+/// assume-false: { __CPROVER_assume(false); }
+///
+/// assert-false-assume-false: {
+///   __CPROVER_assert(false);
+///   __CPROVER_assume(false); }
+///
+/// havoc[,params:<regex>][,globals:<regex>]:
+/// Assign nondet to the targets of pointer-to-non-const parameters
+/// matching regex, and non-const globals matching regex and then
+/// return nondet for non-void functions, e.g.:
+///
+/// int global; const int c_global;
+/// int function(int *param, const int *const_param);
+///
+/// havoc,params:(?!__).*,globals:(?!__).* (where (?!__) means
+/// "not preceded by __", which is recommended to avoid overwrite
+/// internal symbols), leads to
+///
+/// int function(int *param, consnt int *const_param) {
+///   *param = nondet_int();
+///   global = nondet_int();
+///   return nondet_int();
+/// }
+///
+/// nondet-return return nondet for non-void functions
```

Comment:
Nit: `nondet-return:  return nondet for non-void functions` (missing colon separating command line from result)